src/main/java/dev/hephaestus/glowcase/client/render/block/entity/BakedBlockEntityRenderer.java

```diff
@@ -194,103 +183,134 @@ public VertexConsumer getBuffer(RenderLayer layer) {
 			}
 		}
 
-		public void invalidateRegion(BlockPos pos) {
-			// Mark a region as invalid. Before the current set of rebuilding regions (invalid regions from the last frame) have been
-			// built, a RegionBuilder will be created for this region and passed to all BERs to render to
-			invalidRegions.add(new RenderRegionPos(pos));
+		/**
+		 * Causes the render region containing this BlockEntity to be rebuilt -
+		 * do not call this too frequently as it will affect performance.
+		 * An invalidation will not immediately cause the next frame to contain an updated view (and call to renderBaked)
+		 * as all render region rebuilds must call every BER that is to be rendered, otherwise they will be missing from the
+		 * vertex buffer.
+		 */
+		public static void markForRebuild(BlockPos pos) {
+			needsRebuild.add(new RenderRegionPos(pos));
 		}
 
 		// TODO: move chunk baking off-thread?
 
-		private boolean isVisiblePos(RenderRegionPos rrp, Vec3d cameraPos) {
-			return Math.abs(rrp.x - ((int)cameraPos.getX() >> REGION_SHIFT)) <= VIEW_RADIUS && Math.abs(rrp.z - ((int)cameraPos.getZ() >> REGION_SHIFT)) <= VIEW_RADIUS;
+		private static boolean isVisiblePos(RenderRegionPos rrp, Vec3d cam) {
+			return Math.abs(rrp.x - ((int)cam.getX() >> REGION_SHIFT)) <= VIEW_RADIUS && Math.abs(rrp.z - ((int)cam.getZ() >> REGION_SHIFT)) <= VIEW_RADIUS;
 		}
 
-		public void render(MatrixStack matrices, Matrix4f projectionMatrix, Camera camera) {
-			Vec3d cameraPos = camera.getPos();
+		public static void render(WorldRenderContext wrc) {
+			wrc.profiler().push("glowcase:baked_block_entity_rendering");
 
-			// Make builders for invalidated regions
-			for (RenderRegionPos rrp : invalidRegions) {
-				builders.compute(rrp, (k, v) -> {
-					if (v != null) v.reset();
-					else v = new RegionBufferBuilder();
-					return v;
-				});
-			}
-			invalidRegions.clear();
-
-			// Iterate over all RegionBuilders, render and upload to RegionBuffers
-			List<BlockEntity> blockEntities = new ArrayList<>();
-			for (Map.Entry<RenderRegionPos, RegionBufferBuilder> entryBuilder : builders.entrySet()) {
-				RenderRegionPos rrp = entryBuilder.getKey();
-				if (isVisiblePos(rrp, cameraPos)) {
-					// For the current region, rebuild each render layer using the buffer builders
-					// Find all block entities in this region
-					if (currentWorld == null) {
-						break;
-					}
+			Vec3d cam = wrc.camera().getPos();
 
-					for (int chunkX = rrp.x << REGION_FROMCHUNK_SHIFT; chunkX < (rrp.x + 1) << REGION_FROMCHUNK_SHIFT; chunkX++) {
-						for (int chunkZ = rrp.z << REGION_FROMCHUNK_SHIFT; chunkZ < (rrp.z + 1) << REGION_FROMCHUNK_SHIFT; chunkZ++) {
-							blockEntities.addAll(currentWorld.getChunk(chunkX, chunkZ).getBlockEntities().values());
+			if (!needsRebuild.isEmpty()) {
+				wrc.profiler().swap("rebuild");
```

Comment:
This should probably be a `push`, not `swap`, so the path is `glowcase:based_block_entity_rendering.rebuild`

Reply:
👍

Reply:
Done.